docs/pudl_db_index.rst

```diff
@@ -1,5 +1,5 @@
 ===============================================================================
-All PUDL Database Tables
+PUDL Database Index
```

Comment:
Might be more clear to have "table" in there ex: PUDL Database Table Index or something like that

Reply:
Yeah, I went back and forth on that, but then it had both the Tables & Columns, and I didn't want it to seem like it only had Tables. I'll make it more descriptive.